### PR TITLE
ST: Allow users to use any registry for connect build specified via env var

### PR DESCRIPTION
**Pull request by Frawless**
### Type of change

- Bugfix

### Description

This PR enables usage of user-defined registries for KafkaConnect build in systemtest package. 

Resolve https://github.com/strimzi/strimzi-kafka-operator/issues/8277

### Checklist

- [ ] Make sure all tests pass

### CODE REVIEW

File: development-docs/TESTING.md [hunk 1]
```diff
@@ -391,6 +393,13 @@ To set the custom Kafka version in system tests, you need to set the Environment
 If you want to use private registries, before executing the tests, you have to create a secret and then specify the name of the created secret in the env variable called
 `SYSTEM_TEST_STRIMZI_IMAGE_PULL_SECRET` with the container registry credentials to be able to pull images. Note that a secret has to be created in the `default` namespace.
 
+For tests that use `KafkaConnect` build feature, users can use their own registry to where the result image will be pushed.
+The config is specified by two env variables: `CONNECT_BUILD_REGISTRY` and `CONNECT_BUILD_REGISTRY_SECRET`.
+`CONNECT_BUILD_REGISTRY` has to be in format `<REGISTRY>/<ORGANIZATION>`. 
+In case the registry needs authorization, users can specify push-secret by `CONNECT_BUILD_REGISTRY_SECRET`.
+Secret name should reference to k8s secret in `default` namespace.
+In case any of the env vars are not specified, default Minikube or OpenShift registries will be used.
```
[the remark this answers is not in the document]
Comment (Frawless) [reply development-docs/TESTING.md:401]: The image name is static - https://github.com/strimzi/strimzi-kafka-operator/pull/8470/files#diff-0f711d9ed233c37fbe749fd6c4aadce73849f48de3c414d86d9af89d51ea5ef7R470 . To make the whole image name configurable we will need to turn off the parallelism for all connect tests that include connect build and I don't think it makes sense. So at least the tag has to be generated, I can add image-name into the `CONNECT_BUILD_REGISTRY` and maybe change the name of env var.

[the remark this answers is not in the document]
Comment (scholzj) [reply development-docs/TESTING.md:401]: The tag is not really used -> it pushes the image and pulls it through digest. So I think it should mostly work. But I get the point -> however, when you look at how container registries usually work, then you have to setup the image, but the tags are just pushed. So if it was e.g. `quay.io/scholzj/my-custom-build` and you add the tag dynamically, it should work fine as well.

[the remark this answers is not in the document]
Comment (Frawless) [reply development-docs/TESTING.md:401]: Yes, that sounds reasonable. I will update the changes and test them.

[the remark this answers is not in the document]
Comment (scholzj) [reply development-docs/TESTING.md:401]: Thanks.

**Comment by Frawless** [comment]
@strimzi-ci run tests --profile=acceptance --groups=connect

**Comment by Frawless** [comment]
@strimzi-ci run tests --profile=acceptance --groups=connect

**Comment by kornys** [comment]
@strimzi-ci --help

**Comment by kornys** [comment]
@strimzi-ci --help

**Comment by kornys** [comment]
@strimzi-ci help

**Comment by kornys** [comment]
@strimzi-ci --help

**Comment by Frawless** [comment]
@strimzi-ci run tests --profile=acceptance --groups=connect

**Comment by Frawless** [comment]
@strimzi-ci run tests --groups=connect

File: development-docs/TESTING.md [hunk 2]
```diff
@@ -391,6 +393,13 @@ To set the custom Kafka version in system tests, you need to set the Environment
 If you want to use private registries, before executing the tests, you have to create a secret and then specify the name of the created secret in the env variable called
 `SYSTEM_TEST_STRIMZI_IMAGE_PULL_SECRET` with the container registry credentials to be able to pull images. Note that a secret has to be created in the `default` namespace.
 
+For tests that use `KafkaConnect` build feature, users can use their own registry to where the result image will be pushed.
+The config is specified by two env variables: `CONNECT_BUILD_IMAGE_PATH` and `CONNECT_BUILD_REGISTRY_SECRET`.
+`CONNECT_BUILD_IMAGE_PATH` has to be in format `<REGISTRY>/<ORGANIZATION>/<IMAGE_NAME>`.
```
Comment (scholzj) [inline development-docs/TESTING.md:398]: I think it is worth mentioning explicitly that it should be without the tag and that the tag will be appended with the tests to ensure uniqueness.

**Review by scholzj — APPROVED** [review]
One nit to the docs. But LGTM otherwise.

**Pull request merged**